### PR TITLE
I138  leroy - adding 3 pipes

### PR DESCRIPTION
**Pull request by leoRysing**
I've gone ahead and added the code to work the three pipes requested by this issue, which can be found at the bottom of the file

### CODE REVIEW

File: src/shared/pipes.tsx [hunk 1]
```diff
@@ -48,3 +48,16 @@ export const endDatePipe: Function = (startDate: Date, durWeeks: number): string
...
+
+// pipes to add;
+// Empty string pipe
+export const emptyStringPipe: Function = (str: string): string => {
+  return (str === undefined || str === null) ? "" : str;
+};
+
+// EM dash pipe
+export const emDashPipe: Function = (str: string): string => {
+  return (str === undefined || str === null) ? "—" : "str";
```
Comment (jamescd18) [inline src/shared/pipes.tsx:60]: I think you meant the variable str and not the string str?

File: src/shared/pipes.tsx [hunk 2]
```diff
@@ -48,3 +48,16 @@ export const endDatePipe: Function = (startDate: Date, durWeeks: number): string
   endDate.setDate(endDate.getDate() + durWeeks * 7);
   return endDate.toLocaleDateString();
 };
+
+// pipes to add;
+// Empty string pipe
```
Comment (jamescd18) [inline src/shared/pipes.tsx:53]: change to short description of pipe

**Review by jamescd18 — CHANGES_REQUESTED** [review]
Great start! The functionality of the first two pipes look perfect. Here's a few general notes:

There's 3 different parts to ensuring that Prettier is set up properly. If you want, I'd be happy to set up a meeting to help you with the three steps:
- [x] Install the Prettier VScode extension
- [x] Enable format on save
- [x] Set Prettier as the default formatter for typescript files
These are described in more detail in the [onboarding doc here](https://github.com/Northeastern-Electric-Racing/PM-Dashboard-v2/blob/main/docs/Onboarding.md#ide-vscode).

Let me know if you get stuck on the date pipe one, and also add a couple test cases for each pipe in the pipe test file too. Thank you Leroy!

File: src/shared/pipes.tsx (hunk 1 above)
Comment (leoRysing) [reply src/shared/pipes.tsx:60]: Is the convention with pipes to create a variable inside that takes in some data, to then be used to do x task?

Comment (jamescd18) [reply src/shared/pipes.tsx:60]:
```suggestion
  return (str === undefined || str === null) ? "—" : str;
```
Clarified with the above suggestion

File: src/shared/pipes.tsx (hunk 2 above)
Comment (leoRysing) [reply src/shared/pipes.tsx:53]: Will go in an add these changes shortly, today or tomorrow

File: src/shared/pipes.tsx [hunk 3]
```diff
@@ -48,3 +48,20 @@ export const endDatePipe: Function = (startDate: Date, durWeeks: number): string
   endDate.setDate(endDate.getDate() + durWeeks * 7);
   return endDate.toLocaleDateString();
 };
+
+// Returns an empty string if a passed in string is empty, otherwise return the given string
+export const emptyStringPipe: Function = (str: string): string => {
```
Comment (jamescd18) [inline src/shared/pipes.tsx:53]:
```suggestion
export const emptyStringPipe = (str: string): string => {
```

File: src/shared/pipes.tsx [hunk 4]
```diff
@@ -48,3 +48,20 @@ export const endDatePipe: Function = (startDate: Date, durWeeks: number): string
...
+  return (str === undefined || str === null) ? "" : str;
+};
+
+// Similar to the above string; replace an empty string with an EM dash
+export const emDashPipe: Function = (str: string): string => {
```
Comment (jamescd18) [inline src/shared/pipes.tsx:58]:
```suggestion
export const emDashPipe = (str: string): string => {
```

File: src/shared/pipes.tsx [hunk 5]
```diff
@@ -48,3 +48,20 @@ export const endDatePipe: Function = (startDate: Date, durWeeks: number): string
...
+  return (str === undefined || str === null) ? "—" : str;
+};
+
+// return a given data as a string in the local en-US format
+export const datePipe: Function = (date: Date): string => {
```
Comment (jamescd18) [inline src/shared/pipes.tsx:63]:
```suggestion
export const datePipe = (date: Date): string => {
```

File: src/shared/pipes.tsx [hunk 6]
```diff
@@ -48,3 +48,20 @@ export const endDatePipe: Function = (startDate: Date, durWeeks: number): string
...
+  //
+  var theDate = new Date(date);
+  return theDate.toLocaleString('en-US');
+};
```
Comment (jamescd18) [inline src/shared/pipes.tsx:67]:
```suggestion
};

```

File: src/shared/pipes.tsx [hunk 7]
```diff
@@ -48,3 +48,20 @@ export const endDatePipe: Function = (startDate: Date, durWeeks: number): string
...
+
+// Similar to the above string; replace an empty string with an EM dash
+export const emDashPipe: Function = (str: string): string => {
+  return (str === undefined || str === null) ? "—" : str;
+};
+
+// return a given data as a string in the local en-US format
+export const datePipe: Function = (date: Date): string => {
+  //
```
Comment (jamescd18) [inline src/shared/pipes.tsx:64]:
```suggestion
```

File: src/shared/pipes.tsx [hunk 8]
```diff
@@ -48,3 +48,20 @@ export const endDatePipe: Function = (startDate: Date, durWeeks: number): string
   endDate.setDate(endDate.getDate() + durWeeks * 7);
   return endDate.toLocaleDateString();
 };
+
+// Returns an empty string if a passed in string is empty, otherwise return the given string
+export const emptyStringPipe: Function = (str: string): string => {
+  return (str === undefined || str === null) ? "" : str;
+};
+
+// Similar to the above string; replace an empty string with an EM dash
```
Comment (jamescd18) [inline src/shared/pipes.tsx:57]:
```suggestion
// Replace an empty string with an EM dash
```

**Review by jamescd18 — CHANGES_REQUESTED** [review]

File: src/shared/pipes.tsx [hunk 9]
```diff
@@ -48,3 +48,19 @@ export const endDatePipe: Function = (startDate: Date, durWeeks: number): string
...
+// Replace an empty string with an EM dash
+export const emDashPipe = (str: string): string => {
+  return (str === undefined || str === null) ? "—" : str;
+};
+
+// return a given data as a string in the local en-US format
+export const datePipe = (date: Date): string => {
+  var theDate = new Date(date);
+  return theDate.toLocaleString('en-US');
+};
```
Comment (jamescd18) [inline src/shared/pipes.tsx:66]:
```suggestion
};

```

**Review by jamescd18 — CHANGES_REQUESTED** [review]

**Review by jamescd18 — APPROVED** [review]

**Pull request merged**